recipes/quill/all/conanfile.py

```diff
@@ -72,7 +78,8 @@ def _configure_cmake(self):
         self._cmake = CMake(self)
         self._cmake.definitions["QUILL_FMT_EXTERNAL"] = True
         self._cmake.definitions["QUILL_ENABLE_INSTALL"] = True
-
+        self._cmake.definitions["QUILL_USE_BOUNDED_QUEUE"] = self.options.with_bounded_queue
```

Comment:
This option was added in `1.3.2` (https://github.com/odygrd/quill/blob/22d7c4c13fc4dfb1ddad05a7edd8203bd5cb2511/CHANGELOG.md#v132), @uilianries , @SSE4 is it important to add this option only for the versions that have it available? (I'm thinking about the change to the `openssl` where we remove some option based on the version...)

Reply:
It will generate an cmake warning only, but won't affect any behavior. I see no problem keeping it.

Reply:
My concern is not about the CMake variable, but the option itself. There is a bug in Conan related to package-ID computation and how the options from the dependencies affect it. We discovered that problem here (https://github.com/conan-io/examples/pull/61) and it cannot we fixed without breaking existing computation, so it has to wait for Conan 2.0

The problem is that, if we remove option `with_bounded_queue` for older versions (someone could make that PR in the future trying to improve this recipe), we will modify the package ID of the consumers of that recipe-version and they won't have binaries available until we generate it again in Conan Center (this is WIP, not easy to rebuild all the dependencies for every change).

We should have this is mind. It is not a big problem. It is something that happens and we know that Conan Center can have eventually missing packages because of the tons of possible combinations for all the recipes and versions.

Reply:
@jgsogo so, to clarify, should we modify this PR to remove an option? or don't block merge and fix it later with other PR?